include/filewriter/summarywriter_impl.hpp

```diff
@@ -150,6 +150,129 @@ void SummaryWriter<Filewriter>::Image(const std::string& tag,
     remove(fileNames[i].c_str());
 }
 
+template<typename Filewriter>
+void SummaryWriter<Filewriter>::PrCurve(const std::string& tag,
+                                        const std::vector<double>& labels,
+                                        const std::vector<double>& predictions,
+                                        Filewriter& fw,
+                                        int threshold,
+                                        std::vector<double>weights,
+                                        const std::string& displayName,
+                                        const std::string& description)
+{
+    // Pr plugin.
+    mlboard::PrCurvePluginData *prCurvePlugin = new PrCurvePluginData();
+    prCurvePlugin->set_version(0);
+    prCurvePlugin->set_num_thresholds(threshold);
+    std::string pr_curve_content;
+    prCurvePlugin->SerializeToString(&pr_curve_content);
+
+    // PluginMeta data.
+    mlboard::SummaryMetadata_PluginData *pluginData =
+        new SummaryMetadata::PluginData();
+    pluginData->set_plugin_name("pr_curves");
+    pluginData->set_content(pr_curve_content);
+
+    // Summary Meta data.
+    mlboard::SummaryMetadata *metaData = new SummaryMetadata();
+    metaData->set_display_name(displayName == "" ? tag : displayName);
+    metaData->set_summary_description(description);
+    metaData->set_allocated_plugin_data(pluginData);
```

Comment:
```suggestion
    mlboard::SummaryMetadata *metadata = new SummaryMetadata();
    metadata->set_display_name(displayName == "" ? tag : displayName);
    metadata->set_summary_description(description);
    metadata->set_allocated_plugin_data(pluginData);
```

I am sorry I think I myself told you to make this change.